### PR TITLE
[QNN] Refactor fixed point multiplication in requantize

### PR DESCRIPTION
**Pull request by vinx13**
This PR extracts fixed point multiplication in `qnn.requantize` into `util.cc` so that it can be used in other places like `relay::quantize::QuantizeRealize`

Please review @anijain2305 @zhiics

### CODE REVIEW

File: src/relay/qnn/util.cc [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+  tensor =
+      RightShift(tensor, MakeConstantScalar(hp_dtype, total_right_shift));
+
+  return tensor;
+}
+
+}  // namespace qnn
+}  // namespace relay
+}  // namespace tvm
+
```
Comment (anijain2305) [inline src/relay/qnn/util.cc]: Minor - Lets remove the last blank line.

File: src/relay/qnn/util.h [hunk 2]
```diff
@@ -92,7 +93,34 @@ static inline int64_t get_const_int(const tvm::Expr& x) {
...
+ */
+Expr FixedPointMuliply(Expr tensor, double multiplier,
+                       const Array<IndexExpr>& input_shape,
+                       const std::string& rounding);
+
 }  // namespace qnn
 }  // namespace relay
 }  // namespace tvm
 #endif  // TVM_RELAY_QNN_UTIL_H_
+
```
Comment (anijain2305) [inline src/relay/qnn/util.h]: Minor - Lets remove the last blank line.

**Comment by anijain2305** [comment]
Another comment - Currently, QNN includes headers (code) from Relay. I tried to make it as uni-directional as possible. However, this will create bi-directional code inclusion (which might be ok). Therefore, it is worth thinking if we should put FixedPointMultiply in some header in Relay, which QNN can now read from.

**Comment by zhiics** [comment]
LGTM. Or we can just add a util.h header to put the declarations.

**Comment by vinx13** [comment]
@zhiics Can you approve? Or shall we move FixedPointMul to relay/quantize/util.cc ?

**Pull request merged**